File: projects/practica_docker/volumes/java/src/main/java/cat/iesmanacor/gestibgsuite/model/apps/grupscooperatius/ValorItem.java
package cat.iesmanacor.gestibgsuite.model.apps.grupscooperatius;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;

@Entity
@Table(name = "im_app_grups_cooperatius_valor_item")
@EqualsAndHashCode(exclude="item")
public @Data class ValorItem {
    @Id
    @Column(name = "idvalor_item")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idvalorItem;

    @Column(name = "valor", nullable = false, length = 255)
    private String valor;

    @Column(name = "pes", nullable = false, length = 255)
    private Integer pes;

    @ManyToOne(optional = true)
    @JsonManagedReference
    private Item item;
}

File: projects/practica_docker/volumes/quasar/src/model/apps/grupsCooperatius/ValorItem.ts
import {ValorItemMembre} from "src/model/apps/grupsCooperatius/ValorItemMembre";
import {Item} from "src/model/apps/grupsCooperatius/Item";

export interface ValorItem{
  id?: number;
  valor: string;
  pes: number;
  valorsItemMembre?: Array<ValorItemMembre>;
  item: Item;
}
